### PR TITLE
Added support for automatic labeling of PRs.

### PR DESCRIPTION
**Pull request by russjones**
**Description**

Added workflow that automatically labels PRs based on content.

### CODE REVIEW

File: .github/workflows/robot/internal/bot/label.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	return nil
+}
+
+func (b *Bot) label(ctx context.Context) ([]string, error) {
+	// Use a map for deduplication.
+	m := map[string]bool{}
+
+	// The branch name is unsafe, but here we are simply adding a label.
+	if strings.HasPrefix(b.c.Environment.UnsafeBranch, "branch/") {
+		log.Printf("Label: Found backport branch.")
```
Comment (zmb3) [inline .github/workflows/robot/internal/bot/label.go:53]: nit
```suggestion
		log.Println("Label: Found backport branch.")
```

**Review by zmb3 — COMMENTED** [review]
Good start.

In a future iteration it would be nice to remove the special handling for `examples/chart/` and just have a map of path prefixes and labels to apply. Something like:
```
cfg := map[string][]string{
    "lib/srv/desktop": []string{"desktop-access"}
    "examples/chart": []string{"helm", "exmaples"}
}
```

File: .github/workflows/robot/internal/bot/label.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	return out
+}
+
+var prefixes map[string][]string = map[string][]string{
+	"docs/":           []string{"documentation"},
+	"rfd/":            []string{"documentation"},
+	"examples/chart":  []string{"helm"},
+	"lib/srv/desktop": []string{"desktop-access"},
```
Comment (zmb3) [inline .github/workflows/robot/internal/bot/label.go:107]: While we're here, can we add:

```
"lib/srv/desktop/rdp": []string{"rdp"},
"lib/web/desktop.go": []string{"desktop-access"},
"lib/srv/app/":  []string{"application-access"},
"bpf/": []string{"bpf"},
"lib/bpf/": []string{"bpf"},
"tool/tctl/": []string{"tctl"},
"tool/tsh/": []string{"tsh"},
```

File: .github/workflows/robot/internal/bot/label.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	for k, _ := range m {
+		out = append(out, k)
+	}
+
+	return out
+}
+
+var prefixes map[string][]string = map[string][]string{
+	"docs/":           []string{"documentation"},
+	"rfd/":            []string{"documentation"},
```
Comment (zmb3) [inline .github/workflows/robot/internal/bot/label.go:105]:
```suggestion
	"rfd/":            []string{"documentation", "rfd"},
```

**Review by zmb3 — APPROVED** [review]

**Review by fspmarshall — APPROVED** [review]

**Pull request merged**